src/main/java/fr/lunatech/timekeeper/services/imports/ImportUtils.java

```diff
@@ -0,0 +1,356 @@
+        timeSheet.persistAndFlush();
+        return timeSheet.id;
+    }
+
+
+    /**
+     * Returns the clients names from DB.
+     *
+     * @param organizationId as a Long
+     * @return a Set of Client name as String
+     */
+    protected static Set<String> getExistingClientsNames(Long organizationId) {
+        try (final Stream<Client> clientsStream = Client.streamAll()) {
+            return clientsStream
+                    .filter(client -> client.organization.id.equals(organizationId))
+                    .map(client -> client.name.toLowerCase())
+                    .collect(toSet());
+        }
+    }
+
+    /**
+     * Returns the email of a user from DB and the user matching from DB.
+     *
+     * @param organizationId as a Long
+     * @return a Map of an email (String) as a key and a user (User) as value.
+     */
+    protected static Map<String, User> getExistingEmailAndUser(Long organizationId) {
+        try (final Stream<User> userStream = User.streamAll()) {
+            return userStream
+                    .filter(user -> user.organization.id.equals(organizationId))
+                    .collect(Collectors.toMap(user -> user.email, user -> user));
+        }
+    }
+
+    /**
+     * Returns the project name and project matching from DB
+     *
+     * @param organizationId as a Long
+     * @return a Map of project's name (String) as key and project (Project) as value.
+     */
+    protected static Map<String, Project> getExistingProject(Long organizationId) {
+        try (final Stream<Project> projectsStream = Project.streamAll()) {
+            return projectsStream
+                    .filter(project -> project.organization.id.equals(organizationId))
+                    .collect(Collectors.toMap(project -> project.name.toLowerCase(), project -> project));
+        }
+    }
+
+    /**
+     * Returns the clients from DB
+     *
+     * @param organizationId as a Long
+     * @return a List of Client
+     */
+    protected static List<Client> getExistingClients(Long organizationId) {
+        try (final Stream<Client> clientsStream = Client.streamAll()) {
+            return clientsStream
+                    .filter(client -> client.organization.id.equals(organizationId))
+                    .collect(Collectors.toList());
+        }
+    }
+
+    /**
+     * Returns the projects name by clients name from DB.
+     *
+     * @param organizationId as a Long
+     * @return a Map of clients name (String) as a key and a set of projects name (String) as value
+     */
+    protected static Map<String, Set<String>> getExistingProjectsByClient(Long organizationId) {
+        try (final Stream<Project> projectsStream = Project.streamAll()) {
+            return projectsStream
+                    .filter(project -> project.organization.id.equals(organizationId))
+                    .collect(Collectors.groupingBy(p -> p.client.name.toLowerCase(), mapping(p2 -> p2.name.toLowerCase(), toSet())));
```

Comment:
Project without client will trigger a `NullPointerException` here. Either filter project without client OR manage null values in groupingBy.